tomviz/PythonGeneratedDatasetReaction.cxx

```diff
@@ -384,8 +385,175 @@ void PythonGeneratedDatasetReaction::addDataset()
+    QDoubleSpinBox *phi_a3 = new QDoubleSpinBox(&dialog);
+    phi_a3->setValue(0.0);
+    phi_a3->setSingleStep(0.1);
+    parametersLayout->addWidget(phi_a3,4,5,1,1);
+    
+    label = new QLabel("Coma: ", &dialog);
+    parametersLayout->addWidget(label,5,3,1,1);
+    label = new QLabel("Value (nm): ", &dialog);
+    parametersLayout->addWidget(label,5,4,1,1);
+    QDoubleSpinBox *f_c3 = new QDoubleSpinBox(&dialog);
+    f_c3->setRange(-1000000,1000000);
+    f_c3->setValue(1500.0);
+    f_c3->setSingleStep(1000);
+    parametersLayout->addWidget(f_c3,5,5,1,1);
+    label = new QLabel("Orientation (rad): ", &dialog);
+    parametersLayout->addWidget(label,6,4,1,1);
+    QDoubleSpinBox *phi_c3 = new QDoubleSpinBox(&dialog);
+    phi_c3->setValue(0.0);
+    phi_c3->setSingleStep(0.1);
+    parametersLayout->addWidget(phi_c3,6,5,1,1);
+    
+    //Buttons
+    QDialogButtonBox* buttons = new QDialogButtonBox(
+                                                     QDialogButtonBox::Cancel|QDialogButtonBox::Ok, Qt::Horizontal, &dialog);
+    QObject::connect(buttons, SIGNAL(accepted()), &dialog, SLOT(accept()));
+    QObject::connect(buttons, SIGNAL(rejected()), &dialog, SLOT(reject()));
+    
+    layout->addItem(parametersLayout);
+    layout->addWidget(buttons);
+    dialog.setLayout(layout);
+    dialog.layout()->setSizeConstraint(QLayout::SetFixedSize); //Make the UI non-resizeable
+    
+    // substitute values
+    if (dialog.exec() == QDialog::Accepted)
+    {
+      QString pythonScript = this->Internals->scriptSource;
+      pythonScript.replace("###voltage###", QString("voltage = %1").arg(voltage->value()));
+      pythonScript.replace("###alpha_max###", QString("alpha_max = %1").arg(alpha_max->value()));
+      pythonScript.replace("###Nxy###", QString("Nxy = %1").arg(Nxy->value()));
+      pythonScript.replace("###Nz###", QString("Nz = %1").arg(Nz->value()));
+      pythonScript.replace("###dxy###", QString("dxy = %1").arg(dxy->value()));
+      pythonScript.replace("###df_min###", QString("df_min = %1").arg(df_min->value()));
+      pythonScript.replace("###df_max###", QString("df_max = %1").arg(df_max->value()));
+      pythonScript.replace("###c3###", QString("c3 = %1").arg(c3->value()));
+      pythonScript.replace("###f_a2###", QString("f_a2 = %1").arg(f_a2->value()));
+      pythonScript.replace("###phi_a2###", QString("phi_a2 = %1").arg(phi_a2->value()));
+      pythonScript.replace("###f_a3###", QString("f_a3 = %1").arg(f_a3->value()));
+      pythonScript.replace("###phi_a3###", QString("phi_a3 = %1").arg(phi_a3->value()));
+      pythonScript.replace("###f_c3###", QString("f_c3 = %1").arg(f_c3->value()));
+      pythonScript.replace("###phi_c3###", QString("phi_c3 = %1").arg(phi_c3->value()));
+      generator.setScript(pythonScript);
+      const int shape[3] = {Nxy->value(),Nxy->value(),Nz->value()};
+      this->dataSourceAdded(generator.createDataSource(shape));
+    }
+  
+    } //end of else if
```

Comment:
This should be indented only 2 spaces like the opening of this block.  I can fix the indentation stuff though.